### ISSUE TITLE
Mark video as watched after some watching time

### ISSUE THREAD

**Issue by ThatIsAPseudo** [FreeTubeApp/FreeTube#609]
**Is your feature request related to a usage problem (not a bug)? Please describe.**
Whenever I open a video, it's marked as watched, even if I just wanted to see the description, or if I only watched the first seconds of a video before deciding I prefer to watch it later.

**Describe the solution you'd like to see implemented**
Video shouldn't be marked as watched on opening, but after having watched a percentage of the video (choosable in settings)

**Describe alternatives you've considered**
For now I just mark these videos as not watched.

**Comment by GilgusMaximus** [comment]
This could be implemented along #447 . So it would be implemented as a customizable setting that each user can decide for themselves.

**Comment by pawlosck** [comment]
For me, the best option is to manually mark as watched/unwatched independently from time of movie. Longer movies I am not watching one time, only few times, so when after 10 minutes movie will be marked as watched, then I will not know if I watched movie or not.

**Comment by ThatIsAPseudo** [comment]
I agree with the fact that history and watched videos should be treated differently. I guess there could be a "Manual" option if you never want videos to be automatically marked as watched.

**Comment by GilgusMaximus** [comment]
This would be the counter option to 'Do not save history'. However separating watched and history from one another by default is not such a good idea in my opinion, as it would confuse new users and users who have been using it that way all the time.

**Comment by pawlosck** [comment]
We should separate three things (it's only my opinion):
1. History - info, that movie was opened. Maybe with option to add to history after some time. That's all
2. Time of movie - when you stopped watching. Exception to short movies, which should not save time of movie.
3. Status - watched/unwatched. It could be used to filter movies in playlist or in channels with a lot of movies.

Status Watched/Unwatched would be great feature for channels like this (it's not songs, but channel with short informations -> max 90s) -> https://www.youtube.com/channel/UCkcyBx8n6WgN3IqraebaXfQ/videos

> it would confuse new users and users who have been using it that way all the time.
So add option to change this. One user will be use old way and second user change it.
Are you using KDE? It's great GUI because I can change everything, so FreeTube can be similar with a lof of options.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> The generally-accepted number for watch time to count as a view on YouTube is actually a full 30 seconds. 

> What about skipping through a video? If you have a five minute video and a user watches three different ten-second segments of it, it’s possible the view will count; after all, the viewer was watching parts of the video.


What about shorter videos? There are plenty of Vine-style short videos under 30 seconds long. If it takes 30 seconds for a view to count as a view, do 10 second videos never accumulate views? That’s absurd, of course.

Maybe we can use this as baseline

https://growtraffic.com/blog/2017/08/youtube-video-counts-view#What_Makes_a_View_Bad

**Comment by jojo2357** [comment]
I'd like to add that if a video is watched into the SponsorBlock Outro it should be marked as completely watched. Not sure if that deserves its own issue.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing as our behavior is exactly the same as YT here
